tests/pos-macros/i16615.scala

```diff
@@ -0,0 +1,16 @@
+import scala.quoted.*
+
+trait Api:
+  type Reader[E]
+
+def bugImpl[T: Type, Q[_]: Type](using Quotes) =
+  '{
+    val p: Api = ???
+    ${
+      Type.of[p.Reader[T]]
+      Type.of[Q[p.Reader[T]]]
+      Type.of[p.Reader[Q[p.Reader[T]]]]
```

Comment:
We should also test that this work for a `F[_]` that is statically known (right now it does)
```suggestion
      Type.of[p.Reader[Q[p.Reader[T]]]]
      Type.of[List[p.Reader[T]]]
      Type.of[p.Reader[List[p.Reader[T]]]]
      Type.of[p.Reader[List[T]]]
```